domain/schema/modeltriggers_test.go

```diff
@@ -0,0 +1,1012 @@
+
+	_, err = s.DB().Exec(`
+INSERT INTO storage_volume (uuid, volume_id, life_id, provision_scope)
+VALUES (?, ?, 0, 1)
+	`,
+		vUUID.String(), vUUID.String())
+	c.Assert(err, tc.ErrorIsNil)
+
+	return vUUID.String(), vUUID.String()
+}
+
+// newMachineFilesystemAttachment creates a new filesystem attachment that has
+// machine provision scope. The attachment is associated with the provided
+// filesystem uuid and net node uuid.
+func (s *modelStorageSuite) newMachineFilesystemAttachment(
+	c *tc.C, fsUUID string, netNodeUUID string,
+) string {
+	attachmentUUID, err := uuid.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
+
+	_, err = s.DB().ExecContext(
+		c.Context(),
+		`
+INSERT INTO storage_filesystem_attachment (uuid,
+										   storage_filesystem_uuid,
+										   net_node_uuid,
+										   life_id,
+										   provision_scope)
+VALUES (?, ?, ?, 0, 2)
+`,
+		attachmentUUID.String(), fsUUID, netNodeUUID)
+	c.Assert(err, tc.ErrorIsNil)
+
+	return attachmentUUID.String()
+}
+
+// newMachineVolumeAttachment creates a new volume attachment that has
+// machine provision scope. The attachment is associated with the provided
+// volume uuid and net node uuid.
+func (s *modelStorageSuite) newMachineVolumeAttachment(
+	c *tc.C, vUUID string, netNodeUUID string,
+) string {
+	attachmentUUID, err := uuid.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
+
+	_, err = s.DB().ExecContext(
+		c.Context(),
+		`
+INSERT INTO storage_volume_attachment (uuid,
+									   storage_volume_uuid,
+									   net_node_uuid,
+									   life_id,
+									   provision_scope)
+VALUES (?, ?, ?, 0, 2)
+`,
+		attachmentUUID.String(), vUUID, netNodeUUID)
+	c.Assert(err, tc.ErrorIsNil)
+
+	return attachmentUUID.String()
+}
+
+// newMachineVolumeAttachmentPlan creates a new volume attachment plan that has
+// machine provision scope. The attachment is associated with the provided
+// volume uuid and net node uuid.
+func (s *modelStorageSuite) newMachineVolumeAttachmentPlan(
+	c *tc.C, vUUID string, netNodeUUID string,
+) string {
+	attachmentUUID, err := uuid.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
+
+	_, err = s.DB().ExecContext(
+		c.Context(),
+		`
+INSERT INTO storage_volume_attachment_plan (uuid,
+									   		storage_volume_uuid,
+											net_node_uuid,
+											life_id,
+											provision_scope)
+VALUES (?, ?, ?, 0, 2)
+`,
+		attachmentUUID.String(), vUUID, netNodeUUID)
+	c.Assert(err, tc.ErrorIsNil)
+
+	return attachmentUUID.String()
+}
+
+// newModelFilesystemAttachment creates a new filesystem attachment that has
+// model provision scope. The attachment is associated with the provided
+// filesystem uuid and net node uuid.
+func (s *modelStorageSuite) newModelFilesystemAttachment(
+	c *tc.C, fsUUID string, netNodeUUID string,
+) string {
+	attachmentUUID, err := uuid.NewUUID()
+	c.Assert(err, tc.ErrorIsNil)
+
+	_, err = s.DB().ExecContext(
+		c.Context(),
+		`
+INSERT INTO storage_filesystem_attachment (uuid,
+										   storage_filesystem_uuid,
```

Comment:
Is the alignment correct here?

Reply:
no I had tabs instead of spaces. All fixed.